File: src/Starfield2026.BgEditor/backend/src/lib/ohana/Models/PocketMonsters/MM.ts
/**
 * Loads a MM overworld chibi character model from Pokemon.
 * Ported from OhanaCli.Formats.Models.PocketMonsters.MM (C#).
 */

import { BinaryReader } from '../../Core/BinaryReader.js';
import { PkmnContainer } from '../../Containers/PkmnContainer.js';
import { BCH } from '../BCH/BCH.js';
import { OModelGroup } from '../../Core/RenderBase.js';

export class MM {
  static load(data: BinaryReader): OModelGroup {
    const container = PkmnContainer.load(data);
    return BCH.load(BinaryReader.fromBuffer(container.content[0].data!));
  }
}

File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Garc/Models/PocketMonsters/MM.cs
using System.IO;

using MiniToolbox.Garc.Containers;

namespace MiniToolbox.Garc.Models.PocketMonsters
{
    class MM
    {
        /// <summary>
        ///     Loads a MM overworld chibi character model from Pokémon.
        /// </summary>
        /// <param name="data">The data</param>
        /// <returns>The Model group with the character meshes</returns>
        public static RenderBase.OModelGroup load(Stream data)
        {
            RenderBase.OModelGroup models = new RenderBase.OModelGroup();

            OContainer container = PkmnContainer.load(data);
            models = BCH.load(new MemoryStream(container.content[0].data));

            return models;
        }
    }
}
